packages/types/src/primitive/StorageKey.ts

```diff
@@ -26,27 +32,50 @@ export interface StorageFunction {
  */
 export default class StorageKey extends Bytes {
   private _meta?: MetaV4;
+  private _method?: string;
   private _outputType?: string;
+  private _section?: string;
 
   constructor (value?: AnyU8a | StorageKey | StorageFunction | [StorageFunction, any]) {
-    super(StorageKey.decodeStorageKey(value));
+    const { key, method, section } = StorageKey.decodeStorageKey(value);
+
+    super(key);
 
     this._meta = StorageKey.getMeta(value as StorageKey);
+    this._method = method;
     this._outputType = StorageKey.getType(value as StorageKey);
+    this._section = section;
   }
 
-  static decodeStorageKey (value?: AnyU8a | StorageKey | StorageFunction | [StorageFunction, any]): Uint8Array | string | undefined {
-    if (!value || isU8a(value) || isString(value)) {
+  static decodeStorageKey (value?: AnyU8a | StorageKey | StorageFunction | [StorageFunction, any]): Decoded {
+    if (!value || isString(value)) {
```

Comment:
We should probably throw on undefined, just to be sure. Also, isString should be isHex?

Reply:
Since we don't have a hex type and the input is a string, kept to as `isString` - I'd rather let Bytes sort it out for all those cases.

Reply:
`undefined` is the empty case, i.e. `new StorageKey()` - which should just return the default value.